src/cockpit/389-console/src/lib/monitor/replMonitor.jsx

```diff
@@ -231,6 +195,55 @@ export class ReplMonitor extends React.Component {
         if (this.timer) window.clearTimeout(this.timer);
     }
 
+    loadDSRC() {
+        // Load dsrc replication report configuration
+        this.setState({
+            loadingDSRC: true,
+        });
+        const dsrc_cmd = ["dsctl", "-j", this.props.serverId, "dsrc", "display"];
+        log_cmd("loadDSRC", "Check for replication monitor configurations in the .dsrc file", dsrc_cmd);
+        cockpit
+                .spawn(dsrc_cmd, { superuser: true, err: "message" })
+                .done(dsrc_content => {
+                    const content = JSON.parse(dsrc_content);
+                    let credRows = [];
+                    let aliasRows = [];
+                    if ("repl-monitor-connections" in content) {
+                        const report_config = content["repl-monitor-connections"];
+                        for (const [connection, value] of Object.entries(report_config)) {
+                            const conn = connection + ":" + value;
+                            credRows.push(conn.split(':'));
+                            // [repl-monitor-connections]
+                            // connection1 = server1.example.com:389:cn=Directory Manager:*
+                        }
+                    }
+                    if ("repl-monitor-aliases" in content) {
+                        const report_config = content["repl-monitor-aliases"];
+                        for (const [alias_name, value] of Object.entries(report_config)) {
+                            const alias = alias_name + ":" + value;
+                            aliasRows.push(alias.split(':'));
+                            // [repl-monitor-aliases]
+                            // M1 = server1.example.com:38901
+                        }
+                    }
+                    this.setState({
+                        credRows,
+                        aliasRows,
+                        loadingDSRC: false,
+                    });
+                })
+                .fail(err => {
+                    const errMsg = JSON.parse(err);
+                    this.props.addNotification(
+                        "error",
+                        `Failed to get .dsrc information: ${errMsg.desc}`
+                    );
+                    this.setState({
+                        loadingDSRC: false,
+                    });
+                });
+    }
+
     componentDidMount() {
         if (this.state.initCreds) {
```

Comment:
When I go to `Synchronization Report` while there's no replication configured, I get the next error:

```
CMD: ReplMonitor: add credentials during componentDidMount ==> dsconf -j ldapi://%2fvar%2frun%2fslapd-standalone1.socket config get nsslapd-port nsslapd-localhost nsslapd-rootdn

index.js:2 Uncaught TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
    at bV (index.js:2:1754263)
    at Function.<anonymous> (index.js:2:1763262)
    at G (cockpit.js:5:5607)
    at cockpit.js:5:5802
    at k (cockpit.js:5:4658)
```

Reply:
Fixed